src/swift/Sources/APIViewManager.swift

```diff
@@ -0,0 +1,238 @@
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+import AST
+import Parser
+import Source
+
+/// Handles the generation of APIView JSON files.
+class APIViewManager {
+
+    // MARK: Properties
+
+    static var shared = APIViewManager()
+    let args = CommandLineArguments()
+    let tokenFile = TokenFile(name: "", tokens: [], navigation: [])
+
+    // MARK: Methods
+
+    func run() throws {
+        guard let sourcePath = args.source else {
+            SharedLogger.fail("usage error: SwiftAPIView --source PATH")
+        }
+        guard let sourceUrl = URL(string: sourcePath) else {
+            SharedLogger.fail("usage error: `--source PATH` was invalid.")
+        }
+
+        try buildTokenFile(from: sourceUrl)
+
+        let destPath = args.dest ?? "~/Desktop"
+        guard let destUrl = URL(string: destPath) else {
+            SharedLogger.fail("usage error: `--dest PATH` was invalid.")
+        }
+        // TODO: Output the token file
+    }
+
+    func buildTokenFile(from sourceUrl: URL) throws {
+        // TODO: Build AST. Traverse and populate token file.
+        SharedLogger.debug("URL: \(sourceUrl.absoluteString)")
+        let sourceFile = try SourceReader.read(at: sourceUrl.absoluteString)
+        let parser = Parser(source: sourceFile)
+        let topLevelDecl = try parser.parse()
+        let visitor = TokenVisitor()
+        print(try visitor.traverse(topLevelDecl))
+        
+        
+    }
+}
+
+struct TokenNode {
+    var node : ASTNode
+    var tokens : [TokenItem]
+    var depth : Int
+    
+    init(node: ASTNode, tokens: [TokenItem], depth: Int) {
+        self.node = node
+        self.tokens = tokens
+        self.depth = depth
+    }
+}
+
+class TokenVisitor : ASTVisitor {
+    
+    var visited : [ASTNode] = []
+    
+    func registerVisit(to visitingNode : ASTNode) -> Bool {
+        for node in visited {
+            if node.sourceLocation == visitingNode.sourceLocation {
+                return false
+            }
+        }
+        visited.append(visitingNode)
+        return true
+    }
```

Comment:
Instead of saving a huge list of node references, I would recommend a simple dictionary or set where the key is the sourceLocation. In this way you can test membership in constant time, whereas this implementation will be O(n).